apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -67,4 +116,98 @@ defmodule Aecore.Txs.Pool.Worker do
     {:reply, tx_pool, %{}}
   end
 
+  def get_txs_for_address_with_proof(user_txs) do
+    blocks_for_user_txs =
+    for user_tx <- user_txs do
+      get_block_by_txs_hash(user_tx.txs_hash)
+    end
+    merkle_trees =
+    for block <- blocks_for_user_txs do
+      build_tx_tree(block.txs)
+    end
+    user_txs_trees = Enum.zip(user_txs, merkle_trees)
+    proof =
+    for user_tx_tree <- user_txs_trees  do
+      {tx, tree} = user_tx_tree
+      transaction = :erlang.term_to_binary(
+        tx
+        |> Map.delete(:txs_hash)
+        |> Map.delete(:block_hash)
+        |> Map.delete(:block_height)
+        |> Map.delete(:signature)
+        |> TxData.new()
+      )
+      key = :crypto.hash(:sha256, transaction)
+      merkle_proof = :gb_merkle_trees.merkle_proof(key, tree)
+      serialized_merkle_proof = serialize_merkle_proof(merkle_proof, [])
+      Map.put_new(tx, :proof, serialized_merkle_proof)
+    end
+  end
+
+  ## Private functions
+
+  defp split_blocks([block | blocks], address, txs) do
+    user_txs = check_address_tx(block.txs, address, txs)
+    case user_txs do
+      [] -> split_blocks(blocks, address, user_txs)
+      _ ->
+        block_user_txs =
+      for block_user_txs <- user_txs do
+        block_user_txs
+        |>  Map.put_new(:txs_hash, block.header.txs_hash)
+        |>  Map.put_new(:block_hash, block.header.chain_state_hash)
+        |>  Map.put_new(:block_height, block.header.height)
+      end
+        split_blocks(blocks, address, block_user_txs)
+    end
+  end
+
+  defp split_blocks([], address, txs) do
+    txs
+  end
+
+  defp check_address_tx([tx | txs], address, user_txs) do
+    if tx.data.from_acc == address or tx.data.to_acc == address  do
+      user_txs = [
+        Map.from_struct(tx.data)
+        |> Map.put_new(:signature, tx.signature)| user_txs]
+    end
+
+    check_address_tx(txs, address, user_txs)
+  end
+  defp check_address_tx([], address, user_txs) do
+    user_txs
+  end
+
+  defp build_tx_tree(txs) do
```

Comment:
This is a copy of `calculate_root_hash` from `Aecore.Chain.BlockValidation` please merge this functions.